src/transformers/models/whisper/generation_whisper.py

```diff
@@ -606,13 +609,16 @@ def generate(
             condition_on_prev_tokens=condition_on_prev_tokens, generation_config=generation_config
         )
 
-        timestamp_begin = generation_config.no_timestamps_token_id + 1
+        if not is_shortform:
```

Comment:
I see! in practice several tests failed because `generation_config.no_timestamps_token_id` isn't defined for some models. With the prior short form generation these tests passed because `no_timestamps_token_id` wasn't used. 
It could be related to issue #21878. 

Instead of `if not is_shortform:` we could perhaps use: `if generation_config.no_timestamps_token_id is not None:` for now...